File: in_Java/0856 Score of Parentheses.java
class Solution {
    public int scoreOfParentheses(String S) {
        Stack<String> st = new Stack<>();
        for(char c : S.toCharArray()){
            if(c == '(') st.push(Character.toString(c));
            else{
                if(st.peek().equals("(")){
                    st.pop();
                    st.push("1");
                    continue;
                }
                int n = 0;
                while(!st.isEmpty() && !st.peek().equals("(")){
                    n += Integer.parseInt(st.pop());
                }
                st.pop();
                st.push(String.valueOf(n+n));
            }
        }
        int res = 0;
        while(!st.isEmpty()) res += Integer.parseInt(st.pop());
        return res;
    }
}

File: in_Python/0856 Score of Parentheses.py
class Solution(object):
    def scoreOfParentheses(self, S):
        """
        :type S: str
        :rtype: int
        """
        st = []
        for s in S:
            if s == '(':
                st.append(s)
            else:
                if len(st) != 0 and st[-1] == '(':
                    st.pop()
                    st.append(1)
                else:
                    cursum = 0
                    while len(st) != 0 and st[-1] != '(':
                        cursum += st.pop()
                    st.pop()
                    st.append(cursum * 2)
        res = 0
        while len(st) != 0:
            res += st.pop()
        return res
        